python/ray/ml/preprocessor.py

```diff
@@ -37,12 +37,16 @@ def fit(self, dataset: Dataset) -> "Preprocessor":
 
         Fitted state attributes will be directly set in the Preprocessor.
 
+        One is expected to call ``should_fit`` before calling ``fit``.
+
         Args:
             dataset: Input dataset.
 
         Returns:
             Preprocessor: The fitted Preprocessor with state attributes.
         """
+        assert self._is_fittable, "One is expected to call `should_fit` before `fit`."
```

Comment:
Could also make this a no-op when `not self._is_fittable`, which would be more friendly to e.g. chain preprocessors.